integration-tests/src/aave_router.rs

```diff
@@ -0,0 +1,449 @@
+use hydradx_runtime::{AssetRegistry, Omnipool, Stableswap};
+use hydradx_traits::evm::Erc20Encoding;
+use hydradx_traits::evm::EvmAddress;
+use hydradx_traits::router::ExecutorError;
+use hydradx_traits::router::PoolType::Aave;
+use hydradx_traits::router::RouteProvider;
+use hydradx_traits::router::Trade;
+use hydradx_traits::router::{AssetPair, PoolType};
+use hydradx_traits::stableswap::AssetAmount;
+use hydradx_traits::AssetKind;
+use hydradx_traits::Create;
+use orml_traits::MultiCurrency;
+use pallet_asset_registry::Assets;
+use pallet_broadcast::types::Destination;
+use pallet_liquidation::BorrowingContract;
+use pallet_route_executor::TradeExecution;
+use primitives::Balance;
+use sp_runtime::traits::Zero;
+use sp_runtime::DispatchResult;
+use sp_runtime::FixedU128;
+use sp_runtime::Permill;
+use sp_runtime::TransactionOutcome;
+
+fn with_aave(execution: impl FnOnce()) {
+	TestNet::reset();
+	// Snapshot contains the storage of EVM, AssetRegistry, Timestamp, Omnipool and Tokens pallets
+	hydra_live_ext(PATH_TO_SNAPSHOT).execute_with(|| {
+		let pap_contract = EvmAddress::from_slice(hex!("82db570265c37bE24caf5bc943428a6848c3e9a6").as_slice());
+		let pool_contract = crate::liquidation::get_pool(pap_contract);
+		assert_ok!(EVMAccounts::approve_contract(RuntimeOrigin::root(), pool_contract));
+		assert_ok!(Liquidation::set_borrowing_contract(
+			RuntimeOrigin::root(),
+			pool_contract
+		));
+
+		assert_ok!(Currencies::deposit(DOT, &ALICE.into(), 3 * BAG));
+		assert_ok!(EVMAccounts::bind_evm_address(RuntimeOrigin::signed(ALICE.into())));
+
+		let _ = with_transaction(|| {
+			execution();
+			TransactionOutcome::Commit(DispatchResult::Ok(()))
+		});
+	});
+}
+
+fn with_atoken(execution: impl FnOnce()) {
+	with_aave(|| {
+		assert_ok!(Router::buy(
+			hydradx_runtime::RuntimeOrigin::signed(ALICE.into()),
+			DOT,
+			ADOT,
+			BAG,
+			BAG,
+			vec![Trade {
+				pool: Aave,
+				asset_in: DOT,
+				asset_out: ADOT,
+			}]
+		));
+		execution();
+	})
+}
+
+fn with_stablepool(execution: impl FnOnce(AssetId)) {
+	with_atoken(|| {
+		let pool = AssetRegistry::register_sufficient_asset(
+			None,
+			Some(b"pool".to_vec().try_into().unwrap()),
+			AssetKind::StableSwap,
+			Zero::zero(),
+			None,
+			None,
+			None,
+			None,
+		)
+		.unwrap();
+
+		let amplification = 100u16;
+		let fee = Permill::from_percent(1);
+
+		assert_ok!(Stableswap::create_pool(
+			hydradx_runtime::RuntimeOrigin::root(),
+			pool,
+			[DOT, ADOT].to_vec(),
+			amplification,
+			fee,
+		));
+
+		assert_ok!(Stableswap::add_liquidity(
+			hydradx_runtime::RuntimeOrigin::signed(ALICE.into()),
+			pool,
+			BoundedVec::truncate_from(vec![
+				AssetAmount {
+					asset_id: DOT,
+					amount: BAG,
+				},
+				AssetAmount {
+					asset_id: ADOT,
+					amount: BAG,
+				},
+			]),
+		));
+
+		execution(pool);
+	});
+}
+
+const HDX: AssetId = 0;
+const DOT: AssetId = 5;
+const DAI: AssetId = 2;
+const ADOT: AssetId = 1_000_037;
+const ONE: u128 = 1 * 10_u128.pow(10);
```

Comment:
clippy would complain for `1 *`. But i guess we have clippy disabled anyway :D

Reply:
1 is 1

Reply:
fuck clippy